### PR TITLE
Explicitly define `objectWillChange` on `ViewStore`.

### PR DESCRIPTION
**Pull request by nsillik**
Because the `ViewStore` no longer uses `@Published` variables, the conformance to `ObservableObject` no longer sythesizes the `objectWillChange` property on `ViewStore` in iOS 13.0.x, 13.1.x, and some 13.2 versions.

This caused `ViewStore` changes to not propagate, leading to views never updating on these OS versions. This fixes issues #334 and (the already closed) #238.

### CODE REVIEW

**Comment by nsillik** [comment]
I'm also happy to revert the changes to the Case Studies project file, but it's nice being able to test and demo in all supported iOS versions.

File: Examples/CaseStudies/CaseStudies.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1030,6 +1030,7 @@
 				GCC_WARN_UNINITIALIZED_AUTOS = YES_AGGRESSIVE;
 				GCC_WARN_UNUSED_FUNCTION = YES;
 				GCC_WARN_UNUSED_VARIABLE = YES;
+				IPHONEOS_DEPLOYMENT_TARGET = 13.0;
```
Comment (stephencelis) [inline Examples/CaseStudies/CaseStudies.xcodeproj/project.pbxproj:1033]: Let's revert this in favor of running CI against iOS 13.2 (with Xcode 11.2.1).

```suggestion
```

File: Examples/CaseStudies/CaseStudies.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -1083,6 +1084,7 @@
...
```
Comment (stephencelis) [inline Examples/CaseStudies/CaseStudies.xcodeproj/project.pbxproj:1087]: This too.
```suggestion
```

File: Sources/ComposableArchitecture/ViewStore.swift [hunk 3]
```diff
@@ -48,6 +48,10 @@ public final class ViewStore<State, Action>: ObservableObject {
 
   private var viewCancellable: AnyCancellable?
 
+  // N.B. `ViewStore` no longer contains any `@Published` variables, so `objectWillChange`
+  // won't be synthesized automatically. To work around issues on iOS 13 we explicitly declare it.
+  public lazy var objectWillChange = ObservableObjectPublisher()
```
Comment (stephencelis) [inline Sources/ComposableArchitecture/ViewStore.swift:53]: Small suggestion:
```suggestion
  // N.B. `ViewStore` does not use a `@Published` property, so `objectWillChange`
  // won't be synthesized automatically. To work around issues on iOS 13 we explicitly declare it.
  public private(set) lazy var objectWillChange = ObservableObjectPublisher()
```

**Review by stephencelis — APPROVED** [review]
Thanks!

**Pull request merged**